opensrp-reveal/src/main/java/org/smartregister/reveal/sync/RevealClientProcessor.java

```diff
@@ -165,28 +153,61 @@ private String processSprayEvent(Event event, ClientClassification clientClassif
-            // todo: add logic to update business status when data dictionary is available
+            operationalAreaId = updateTask(event, localEvents, MOSQUITO_COLLECTION);
+
             try {
-                // todo: add status when data dictionary specifies how to collect and determine its value
                 String startDate = event.findObs(null, false, JsonForm.TRAP_SET_DATE).getValue().toString();
                 String endDate = event.findObs(null, false, JsonForm.TRAP_FOLLOW_UP_DATE).getValue().toString();
                 event.getDetails().put(START_DATE, startDate);
                 event.getDetails().put(END_DATE, endDate);
+
+                RevealClientProcessor clientProcessor = getInstance(RevealApplication.getInstance().getApplicationContext());
+                event.getDetails().put(STATUS, clientProcessor.calculateBusinessStatus(event, MOSQUITO_COLLECTION));
+
                 Client client = new Client(event.getBaseEntityId());
                 processEvent(event, client, clientClassification);
-                if (localEvents) {
-                   // todo: may need to add extra task processing for local events vs remote
-                    Log.e(TAG, "Processing local events");
-                }
             } catch (Exception e) {
                 Log.e(TAG, "Error processing spray event", e);
             }
         }
         return operationalAreaId;
     }
 
-    public String calculateBusinessStatus(Event event) {
-        Obs businessStatusObs = event.findObs(null, false, JsonForm.TASK_BUSINESS_STATUS);
+    private String updateTask(Event event, boolean localEvents, String interventionType) {
+        String taskIdentifier = event.getDetails().get(TASK_IDENTIFIER);
+        Task task = taskRepository.getTaskByIdentifier(taskIdentifier);
+        String operationalAreaId = null;
+        if (task != null) {
+            task.setBusinessStatus(calculateBusinessStatus(event, interventionType));
+            task.setStatus(Task.TaskStatus.COMPLETED);
+            // update task sync status to unsynced if it was already synced,
+            // ignore if task status is created so that it will be created on server
+            if (localEvents && BaseRepository.TYPE_Synced.equals(task.getSyncStatus())) {
+                task.setSyncStatus(BaseRepository.TYPE_Unsynced);
+            } else if (!localEvents) {
+                // for events synced from server and task exists mark events as being fully synced
+                eventClientRepository.markEventAsSynced(event.getFormSubmissionId());
+            }
+            taskRepository.addOrUpdate(task);
+            operationalAreaId = task.getGroupIdentifier();
+        } else {
+            eventClientRepository.markEventAsTaskUnprocessed(event.getFormSubmissionId());
+        }
+        return operationalAreaId;
+    }
+
+    public String calculateBusinessStatus(Event event, String interventionType) {
```

Comment:
This should be unified because business status should be calculates on form, we should not calculate on code so that its possible to have different calculations per country

Reply:
Done.